src/core/manager.c

```diff
@@ -940,15 +940,15 @@ int manager_new(RuntimeScope runtime_scope, ManagerTestRunFlags test_run_flags,
         if (r < 0)
                 return r;
 
-        r = hashmap_ensure_allocated(&m->units, &string_hash_ops);
+        r = hashmap_ensure_allocated(&m->units, &string_hash_ops_free);
```

Comment:
Please don't search and replace this. For these hashmaps/sets, the lifetime of values and bound to `Unit` objects, and `manager_free` does not free the values, as they aren't owned by the `Manager` object.